packages/dds/tree/src/test/simple-tree/schemaFactory.spec.ts

```diff
@@ -101,8 +101,38 @@ describe("schemaFactory", () => {
 	it("instanceof", () => {
 		const schema = new SchemaFactory("com.example");
 
-		const config = new TreeConfiguration(schema.number, () => 5);
+		class A extends schema.object("A", {}) {}
+		class B extends schema.object("B", {}) {}
+		const C = schema.object("C", {});
+		const StructuralArray = schema.array(A);
+		const NominalArray = schema.array("D", A);
+
+		const a = new A({});
+		assert(a instanceof A);
+		assert(a instanceof TreeNode);
+		assert(!(a instanceof B));
 
+		const c = new C({});
+		assert(c instanceof C);
+		assert(c instanceof TreeNode);
+		assert(!(c instanceof B));
+
+		const n = new NominalArray([]);
+		assert(n instanceof NominalArray);
+		assert(n instanceof TreeNode);
+		assert(!(n instanceof B));
```

Comment:
why is the check against B interesting? I would think C would be more interesting? or StructuralArray?

Reply:
instanceof comparing to structurally named schema does not compile as we do not expose those schema as classes (thus they don't publicly extend TreeNode).

Mostly this is just testing some negative case, ensuring that the implementation does not fail (ex: loop forever scanning prototypes, or return true). It doesn't really matter what its testing against since in the negative case the type we are testing against isn't used in any way except for comparing its prototype.